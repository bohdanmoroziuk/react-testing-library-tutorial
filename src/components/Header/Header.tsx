import 'components/Header/Header.css';

export interface HeaderProps {
  title: string;
}

export default function Header({ title }: HeaderProps) {
  return (
    <h1 title="Header" className="header">{title}</h1>
  );
}
