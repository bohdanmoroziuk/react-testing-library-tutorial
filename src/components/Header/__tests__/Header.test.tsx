import { render, screen } from '@testing-library/react';

import Header from 'components/Header/Header';

describe('<Header />', () => {
  test('renders successfully', () => {
    render(<Header title="Todo" />);

    const target = screen.getByTitle('Header');

    expect(target).toBeTruthy();
    expect(target).toBeInTheDocument();
    expect(target).toMatchSnapshot();
  });

  test('renders same text passed into title prop', () => {
    const title = "Followers";

    render(<Header title={title} />)

    const target = screen.getByTitle('Header');

    expect(target.textContent).toBe(title);
    expect(target).toHaveTextContent(title);
  });
});
