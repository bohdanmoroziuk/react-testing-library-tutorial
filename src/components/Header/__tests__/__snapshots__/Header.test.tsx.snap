// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders successfully 1`] = `
<h1
  class="header"
  title="Header"
>
  Todo
</h1>
`;
